feature: 매칭 현황 기능 구현 및 정책정의서 반영

Lee-Dahyeon opened the pull request:
## ⚠️ 연관된 이슈 번호 #62 
- close #62 

## 📋 작업 내용(수정하겠습니다!)
- 모든 테이블에 is-deleted컬럼 추가하여 soft delete 진행
- Chatting enum 칼럼 추가
- 매칭 신청에 따른 메시지 저장

## 📷 스크린샷(선택) 

## 💬 리뷰 요구사항
> 리뷰어가 특별히 봐주었으면 하는 부분이 있다면 작성해주세요.
> e.g. 메서드 XXX의 이름을 더 잘 짓고 싶은데 혹시 좋은 명칭이 있을까요?

Lee-Dahyeon committed:
feature: USER_WRONG_PASSWORD 에러 메시지 수정

Lee-Dahyeon committed:
feature: 채팅 입력 시간에 따른 chatRoom 칼럼 업데이트
- chatRoom 속 updateChatTime 메서드 추가
- ChattingService 속 updateChatTime 메서드 반영

Lee-Dahyeon committed:
feature: 매칭 현황 조회
- ParticipantService와 contoller 로직 추가

Lee-Dahyeon committed:
feature: GetMatchStatusResponse dto 생성

Lee-Dahyeon committed:
feature: Chatting enum 칼럼 추가
- enum 타입의 MessageType 생성 (SYSTEM, TEXT)

Lee-Dahyeon committed:
feature: ParticipantService 추가 수정
- getMatchResult 로직 수정
- 에러 처리 수정(get()->orElseThrow())
- @EventPublish 어노테이션 추가

Lee-Dahyeon committed:
feature: Chatting 칼럼 수정
- messageType ColumnDefault "TEXT"로 수정

Lee-Dahyeon committed:
feature: GetMatchStatusResponse 필드명 수정
- owner -> myself

Lee-Dahyeon committed:
feature: BaseEntity 생성 및 적용

Lee-Dahyeon committed:
feature: 게임 등록 취소 기능 구현

Lee-Dahyeon committed:
feature: 게임방 목록 조회 기준 추가
- isDeleted가 false인 게임방 기준 추가

Lee-Dahyeon committed:
feature: 매칭 확정 문구
- 참가자 매칭 신청 및 확정 메시지 저장 로직 추가

Lee-Dahyeon committed:
feature: chatting 엔티티 허용
- 메시지 타입이 SYSTEM인 경우는 User가 없기 때문에 User null 허용으로 수정

Lee-Dahyeon committed:
feature: 채팅중 응답 값 수정
- 와이어프레임을 반영하여 시간은 삭제, 메시지타입을 전송

Lee-Dahyeon committed:
feature: 채팅 메시지 타입 추가 저장
- 채팅과정에서의 메시지는 TEXT 타입으로 메시지를 저장하는 것으로 수정

Lee-Dahyeon committed:
feature: chatRoomService파일 삭제에 따른 임포트 제거

LJH098 commented on src/main/java/com/hyunsolution/dangu/chatting/domain/Chatting.java:
```diff
@@ -42,10 +46,16 @@ public class Chatting {
     @CreatedDate
     private LocalDateTime createdAt;
 
+    @Enumerated(EnumType.STRING)
+    @Column(nullable = false)
+    @ColumnDefault("TEXT")
+    private MessageType messageType;
+
```
MessageType은 왜 기본값 설정을 하신건가요?

Lee-Dahyeon replied:
기본값을 설정하여 다른 코드에서 default값인 TEXT인 경우는 삭제하였습니다!

LJH098 commented on src/main/java/com/hyunsolution/dangu/participant/service/ParticipantService.java:
```diff
@@ -68,8 +75,13 @@ public void updateMatching(Long id, Long chatRoomId, UpdateParticipantMatchReque
+
         // workspaceId 변수 저장
-        Long workspaceId = chatRoomRepository.findById(chatRoomId).get().getWorkspace().getId();
+        Long workspaceId = chatRoomRepository.findById(chatRoomId).orElseThrow(() -> ChatRoomNotFoundException.EXCEPTION).getWorkspace().getId();
```
...??!! 
chatRoomRepository.findById(chatRoomId).orElseThrow(() -> ChatRoomNotFoundException.EXCEPTION) 을 중복 작성하고 있습니다.

LJH098 commented on src/main/java/com/hyunsolution/dangu/participant/service/ParticipantService.java:
```diff
@@ -68,8 +75,13 @@ public void updateMatching(Long id, Long chatRoomId, UpdateParticipantMatchReque
         participant.updateParticipantMatch(request.isMatch());
 
+        //참가자 매칭 신청 메시지 저장
+        String uid= participantRepository.findById(id).orElseThrow(() -> ParticipantNotFoundException.EXCEPTION).getUser().getUid();
```
예..? 여기서 id는 userId인데 participantRepository.findById(id)를 하신다구요..?!
findById를 하면 안될 뿐더러 단순히 uid를 찾는거면 userId로 user를 조회하는게 더 좋지않을까요?

LJH098 commented on src/main/java/com/hyunsolution/dangu/participant/service/ParticipantService.java:
```diff
@@ -102,7 +119,27 @@ private void finalizeWorkspaceMatching(Long workspaceId) {
+        List<Participant> participants = participantRepository.findByChatRoomId(chatRoomId);
+
+        //자신의 매칭 신청 현황
+        boolean myself = participantRepository.findById(userId).orElseThrow(() -> UserNotFoundException.EXCEPTION).isParticipantMatch();
```
... ㅜㅜ 왜자꾸 userId로 participantRepository.findById 하는거죠

LJH098 commented on src/main/java/com/hyunsolution/dangu/participant/service/ParticipantService.java:
```diff
@@ -102,7 +119,27 @@ private void finalizeWorkspaceMatching(Long workspaceId) {
+        boolean counterpart = false;
+        for (Participant participant : participants) {
+            //id와 해당 채팅방에 있는 사람들을 비교 같으면 owner
+            if (!participant.getId().equals(userId)) {
+                counterpart = participant.isParticipantMatch();
+            }
+        }
```
.... 왜 participant의 pk와 user pk랑 비교하나요..

Lee-Dahyeon committed:
feature: spotless 적용

Lee-Dahyeon committed:
feature: ParticipantRepository 수정
- findByUserIdInParticipantsId 추가 (채팅방 참여자 내 클릭누른 사용자 추출)

Lee-Dahyeon committed:
feature: ParticipantService 수정
- userId->participantId를 사용하도록 수정
- 자신의 매칭 신청 로직 수정
  - 채팅방 참여자의 아이디 리스트로 가져와 db에서 사용자의 id와 동일한 Participant의 id 반환
